fcli-other/fcli-functional-test/src/ftest/groovy/com/fortify/cli/ftest/fod/FoDAppSpec.groovy

```diff
@@ -6,21 +6,145 @@ import com.fortify.cli.ftest._common.Fcli
 import com.fortify.cli.ftest._common.spec.FcliBaseSpec
 import com.fortify.cli.ftest._common.spec.FcliSession
 import com.fortify.cli.ftest._common.spec.Prefix
+import com.fortify.cli.ftest.fod._common.FODAppRel
+import com.fortify.cli.ftest.ssc._common.SSCAppVersion
 
+import spock.lang.AutoCleanup
+import spock.lang.Shared
+import spock.lang.Stepwise
 import spock.lang.Unroll
 
-@Prefix("fod.app") @FcliSession(FOD)
+@Prefix("fod.app") @FcliSession(FOD) @Stepwise
 class FoDAppSpec extends FcliBaseSpec {
-    @Unroll
+    @Shared FODAppRel webApp = null;
+    @Shared FODAppRel mobileApp = null;
+    @Shared FODAppRel microservicesApp = null;
+    
     def "list"() {
         def args = "fod app list"
         when:
             def result = Fcli.run(args)
         then:
             verifyAll(result.stdout) {
-                // TODO Add expectations
+                size()>=1
+                if(size()>1) {
+                it[0].replace(' ', '').equals("IdNameTypeCriticality")
+                } else {
+                it[0].equals("No data")
+                }
             }
-        where:
-            i << (1..5)
     }
-}
\ No newline at end of file
+    
+    def "createWebApp"() {
+        when:
+            webApp = new FODAppRel().createWebApp();
+        then:
+            noExceptionThrown()
+    }
+    
+    def "createMicroserviceApp"() {
+        when:
+            microservicesApp = new FODAppRel().createMicroservicesApp();
+        then:
+            noExceptionThrown()
+    }
+    
+    def "createMobileApp"() {
+        when:
+            mobileApp = new FODAppRel().createMobileApp();
+        then:
+            noExceptionThrown()
+    }
+    
+    def "verifyCreated"() {
+        def args = "fod app list --store apps"
+        when:
+            def result = Fcli.run(args)
+        then:
+            verifyAll(result.stdout) {
+                it.any { it.contains(webApp.appName) }
+                it.any { it.contains(microservicesApp.appName) }
+                it.any { it.contains(mobileApp.appName) }
+            }
+    }
+    
+    def "get.byId"() {
+        def args = "fod app get ::apps::get(0).applicationId"
```

Comment:
Why are you using the first record from the list command, rather than the explicitly created web/mobile/microservices app? That would allow you to verify that the `get` command returns the expected `appName` for example.

Reply:
because it doesnt matter at all. the creation of all 3 apps is verified in the verifyCreated step

Reply:
The id is available through the `FoDAppRel::get` method, right? You're currently using the same variable-based approach to get the id of the first record returned by the list command. As mentioned, getting the details of the app that has just been created allows for additional expectations to be verified, like that the get command returns the expected application name.